yarn-project/end-to-end/scripts/native-network/validators.sh

```diff
@@ -36,7 +36,15 @@ if [ "$NUM_VALIDATORS" -eq 1 ]; then
   echo "Running single validator directly"
   eval "${CMD[0]}"
 else
-  echo "Running $NUM_VALIDATORS validators interleaved"
-  # Execute the run_interleaved.sh script with the commands
-  "$(git rev-parse --show-toplevel)/scripts/run_interleaved.sh" "${CMD[@]}"
+  echo "Running $NUM_VALIDATORS validators sequentially, interleaved"
+  FIRST_PORT=8081
+
+  # check if we're running against anvil
+  if curl -s -H "Content-Type: application/json" -X POST --data '{"method":"web3_clientVersion","params":[],"id":49,"jsonrpc":"2.0"}' $ETHEREUM_HOST | jq .result | grep -q anvil; then
+    "$(git rev-parse --show-toplevel)/scripts/run_interleaved.sh" "${CMD[@]}"
+  else
+    # Use run_interleaved with a wait condition
+    WAIT_CONDITION="curl -s http://127.0.0.1:$FIRST_PORT/status >/dev/null"
```

Comment:
was this wait condition added to prevent them being added with the same key? https://github.com/AztecProtocol/aztec-packages/pull/10327 this pr pre adds the validators at contract deployment time so may invalidate the need for this work around